### ISSUE TITLE
Terrible Results in T2I Fine-tuining.

### ISSUE THREAD

**Issue by dayunyan** [ByteDance-Seed/Bagel#192]
After my tests, I discovered a very strange issue. 
First, I fine-tuned on my T2I dataset. Below is the script I used for fine-tuning.
```bash
torchrun \
  --nnodes=1 \
  --node_rank=0 \
  --nproc_per_node=8 \
  --master_addr=127.0.0.1 \
  --master_port=12345 \
  train/pretrain_unified_navit.py \
  --dataset_config_file ./data/configs/t2i.yaml \
  --layer_module Qwen2MoTDecoderLayer \
  --max_latent_size 64 \
  --model_path $model_path \
  --use_flex True \
  --resume-from $model_path \
  --finetune_from_hf True \
  --auto_resume True \
  --resume-model-only True \
  --finetune-from-ema True \
  --num_shard 8 \
  --wandb_offline True \
  --results_dir $output_path \
  --checkpoint_dir $ckpt_path \
  --log_every 10 \
  --save_every 1000 \
  --max_save_num 3 \
  --lr 2e-6 \
  --total_steps 20000 \
  --visual_und False\
  --expected_num_tokens 10240 \
  --max_num_tokens 11520 \
  --max_num_tokens_per_sample 10240 \
  --num_workers 1
```
According to the training log, I found that the model experienced a sharp increase in loss around step 5000, which seemed to indicate a severe problem of knowledge forgetting. 
![Image](https://github.com/user-attachments/assets/92443a1d-4e4c-41d3-a604-df18e95901c6)
The following is a comparison of the quantitative results between the fine-tuned model and the zero-shot model.
![Image](https://github.com/user-attachments/assets/c7fef3e3-f7e0-4cbd-a240-7e422a9aef5c)

Then, since the ckpt saved during training lacked the vit-related components (#160 ), I supplemented the vit components from the official model to the ckpt saved during training based on the script provided by @AbhinavJangra29 (🤗THANKS)
. After my tests, I found that the model with the supplemented vit performed slightly worse in inference results compared to the directly saved model. The quantitative results are as follows:
![Image](https://github.com/user-attachments/assets/745d8545-f66d-4241-95e4-4dae62a01bcb)

**Comment by dayunyan** [comment]
Besides, I can guarantee that it's not a problem with the data. This data has been verified through multiple practical tests.

**Comment by Andy1621** [comment]
Hi! Based on your log, why are the `total_samples` and `total_mse_tokens` larger in 0~5k where the `mse_loss` decreases and `total_norm` increases?

**Comment by dayunyan** [comment]
> Hi! Based on your log, why are the `total_samples` and `total_mse_tokens` larger in 0~5k where the `mse_loss` decreases and `total_norm` increases?

I checked the dataloader part. My data structure is as follows: 
![Image](https://github.com/user-attachments/assets/dfc9a54e-d331-4ae4-b27e-38f8ac62879d)
data config is as follows:
```yaml
t2i_pretrain:
  dataset_names:
  - 01_chinese_home_style_dishes
  - 02_cartoon_illustration
  - 03_fruits
  - 04_hotel
  - 05_red_envelope
  - 06_golden_coins
  - 07_handsome_man
  - 08_scenery
  - 09_general_scene
  - 10_handsome_godman
  - 17_e_en
  image_transform_args:
    image_stride: 16
    max_image_size: 1024
    min_image_size: 512
  is_mandatory: true
  num_used_data:
  - 6
  - 16
  - 6
  - 6
  - 4
  - 1
  - 31
  - 7
  - 313
  - 299
  - 290
  weight: 1
```
datainfo is as follows:
```python
't2i_pretrain': {
        '01_chinese_home_style_dishes': {
            'data_dir': 'path/to/dataset',
            'num_files': 0, # number of data units to be sharded across all ranks and workers
            'num_total_samples': 0,
        },
...
}
```
I wonder if it's because of the uneven data that the total_sample between 0 and 5k is larger?

**Comment by Andy1621** [comment]
Are images with different ratios/sizes are randomly shuffle in all parquets?

**Comment by dayunyan** [comment]
> Are images with different ratios/sizes are randomly shuffle in all parquets?

Yes, the size differences between different categories of images may be quite large, but the ratios are all random.

Should I randomly mix all these categories into one large dataset instead of having one dataset for each category as I do now?
